presto-client/src/main/java/com/facebook/presto/client/StatementClient.java

```diff
@@ -374,6 +375,13 @@ private RuntimeException requestFailedException(String task, Request request, Js
     {
         gone.set(true);
         if (!response.hasValue()) {
+            if (response.getException() != null) {
+                for (Throwable throwable : getCausalChain(response.getException())) {
```

Comment:
Why are you discarding exception chain up to (excluding) first `ClientException`? (I'm worried that discarding exceptions will hurt problem diagnosability)

Can it be that `ClientException` is "buried"? 
If so, what about `return new ClientException(...., response.getException())` when causal chain contains a `ClientException`?

Reply:
This whole fix is really hacky. The `fixValue()` method is called deep inside Jackson serialization, so the `ClientException` will be wrapped several times. We throw `ClientException` when we know that it is the real error that should be presented to the user.

Reply:
Appreciated. Maybe leave this info as a code comment?